src/map/script.cpp

```diff
@@ -6659,6 +6659,88 @@ BUILDIN_FUNC(getelementofarray)
 	return SCRIPT_CMD_SUCCESS;
 }
 
+/// return the matches of an array, from the starting index.
+/// ex: searcharray arr,1,2,3;
+///
+/// searcharray <array variable>,<value1>{,<value2>...};
+BUILDIN_FUNC(searcharray)
+{
+	struct script_data *data;
+	const char* name;
+	int id , i, j, array_size , case_count = 0;
+	struct map_session_data* sd = NULL;
+	data = script_getdata(st, 2);
+	
+	if (!data_isreference(data))
+	{
+		ShowError("script:searcharray: not a variable\n");
+		script_reportdata(data);
+		script_pushnil(st);
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	id = reference_getid(data);
+	name = reference_getname(data);
+
+	if (not_server_variable(*name)) {
+		if (!script_rid2sd(sd))
```

Comment:
Merge those two ifs together please and change the return code to SCRIPT_CMD_FAILURE.